tests/drivers/uart/uart_async_api/src/test_uart_async.c

```diff
@@ -132,7 +132,9 @@ void test_chained_read(void)
 		uart_tx(uart_dev, tx_buf, sizeof(tx_buf), 100);
 		zassert_equal(k_sem_take(&tx_done, 100), 0, "TX_DONE timeout");
 		zassert_equal(k_sem_take(&rx_rdy, 1000), 0, "RX_RDY timeout");
-		zassert_equal(read_len, sizeof(tx_buf),
+		size_t read_len_temp = read_len;
```

Comment:
Just curious; why didn't you just make read_len non-volatile?

Reply:
read_len is a global variable checked in loop and changed in interrupt. Is there a way to make it work without using volatile with enabled optimization?

Reply:
Nope, I think this should be the right solution